src/lib/viewers/image/__tests__/ImageViewer-test.js

```diff
@@ -576,4 +606,55 @@ describe('lib/viewers/image/ImageViewer', () => {
             expect(image.imageEl.src).toBe(url);
         });
     });
+
+    describe('getViewportDimensions', () => {
+        test('should return width and height', () => {
+            image.wrapperEl = document.createElement('img');
+            Object.defineProperty(image.wrapperEl, 'clientWidth', { value: 100 });
+            Object.defineProperty(image.wrapperEl, 'clientHeight', { value: 100 });
+
+            const result = image.getViewportDimensions();
+
+            expect(result).toEqual({ width: 70, height: 70 });
+        });
+    });
+
+    describe('handleZoomEvent', () => {
+        let height;
+        let width;
+
+        beforeEach(() => {
+            image.wrapperEl = document.createElement('img');
+            Object.defineProperty(image.wrapperEl, 'clientWidth', { value: 100 });
+            Object.defineProperty(image.wrapperEl, 'clientHeight', { value: 100 });
+            jest.spyOn(image, 'processAnnotationModeChange');
+        });
+
+        test.each`
+            currentState                   | height | width  | should
+            ${AnnotationState.REGION}      | ${60}  | ${60}  | ${'image does not overflow the viewport'}
+            ${AnnotationState.REGION}      | ${110} | ${110} | ${'image does overflow the viewport'}
+            ${AnnotationState.REGION_TEMP} | ${110} | ${110} | ${'image does overflow the viewport'}
+        `('should not call processAnnotationImage if $should and currentState is $currentState', () => {
+            image.annotationControlsFSM.currentState = AnnotationState.REGION;
```

Comment:
Instead of setting `currentState`, we could instead instantiate a new instance with `initialState`